src/Reader.php

```diff
@@ -329,19 +317,23 @@ protected function isValidKey($value)
     protected function getRow($offset)
     {
         $fileObj = $this->getIterator();
-        $fileObj->setFlags(SplFileObject::READ_AHEAD | SplFileObject::SKIP_EMPTY);
         $iterator = new LimitIterator($fileObj, $offset, 1);
         $iterator->rewind();
-        $line = $iterator->current();
-
-        if (empty($line)) {
+        $row = $iterator->current();
+        if (empty($row)) {
             throw new InvalidArgumentException('the specified row does not exist or is empty');
         }
 
-        if (0 === $offset && $this->isBomStrippable()) {
-            $line = mb_substr($line, mb_strlen($this->getInputBOM()));
+        if (0 !== $offset || !$this->isBomStrippable()) {
+            return $row;
+        }
+
+        $bom_length = mb_strlen($this->getInputBOM());
+        $row[0] = mb_substr($row[0], $bom_length);
+        if ($this->enclosure == mb_substr($row[0], 0, 1) && $this->enclosure == mb_substr($row[0], -1, 1)) {
+            $row[0] = mb_substr($row[0], 1, -1);
         }
 
-        return str_getcsv($line, $this->delimiter, $this->enclosure, $this->escape);
+        return $row;
```

Comment:
The expected return value for this method is an array, and now appears to be a string.